src/entity/object/FallingBlock.php

```diff
@@ -153,4 +160,19 @@ protected function syncNetworkData(EntityMetadataCollection $properties) : void{
 	public function getOffsetPosition(Vector3 $vector3) : Vector3{
 		return $vector3->add(0, 0.49, 0); //TODO: check if height affects this
 	}
+
+	protected function onHitGround() : ?float{
+		if($this->getBlock()->getId() === VanillaBlocks::ANVIL()->getId()) {
+			foreach($this->getWorld()->getCollidingEntities($this->getBoundingBox()) as $ent){
+				if($ent instanceof Living){
+					$lastCause = $ent->lastDamageCause;
+					if(!($lastCause instanceof EntityDamageByBlockEvent) || $lastCause->getDamager() !== $this->getBlock()){
+						$damageSource = new EntityDamageByBlockEvent($this->getBlock(), $ent, EntityDamageEvent::CAUSE_FALLING_BLOCK, min($this->fallTime / 2, 40));
```

Comment:
I'm not completely sure `EntityDamageByBlockEvent` suits here, since it's actually the entity doing the damage.